activerecord/lib/active_record/connection_adapters/abstract/connection_handler.rb

```diff
@@ -322,7 +322,7 @@ def disconnect_pool_from_pool_manager(pool_manager, role, shard)
         #
         def resolve_pool_config(config, connection_name, role, shard)
           db_config = Base.configurations.resolve(config)
-
+          db_config.validate
```

Comment:
CI caught this. The `ridiculous` adapter from another test was passed into `#establish_connection` and a pool was created with this config, but then every time it was attempted to be used it raised because the adapter was invalid.

Here, since we delay validating the connection to an initializer, and not when the class is initialized, it's possible to have invalid connection configs. I bet we have other bugs of this class we haven't caught yet.

Reply:
Is this _sufficient_? (As in, without the initializer?)

I like the idea of proactively validating things while we're setting up the pool, rather than only noticing a problem when we try to use the config to actually connect... but while the initializer definitely does that, I worry that it may run into #50086-ish issues with config keys that aren't intended to be proper valid configs at all. Noticing a problem as soon as you tell us that you're using a given key as a connection [but not before] seems ideal, maybe?

Reply:
Oooo that's a very good question. It might be! I think you're right. I'll remove the initializer and see.

Unfortunately though, moving the validation up into the pool invalidates what I was trying to fix originally. The build is green, but the pool is configured and validates before the `on_load(:active_record)` defined in app initializers runs. The repro steps above now fail.

I think I'll try to design an integration test that tests this rather than manual reproduction.

But. It sends me back to how to sequence this such that you can register connection adapters before something validates the adapter in initialization but after Active Record is loaded.